### PR TITLE
 Implementation of DataArts Factory resources API

### PR DESCRIPTION
**Pull request by sgmv**
### What this PR does / why we need it

The next changes for DataArt API. This PR brings the ability to manage DataArts Factory resources.

### CODE REVIEW

File: acceptance/openstack/dataarts/v1/resources_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+const resourceName = "testResource"
+const bucketName = "dataart-test-bucket"
+const fileName = "testFile.txt"
+
+func TestDataArtsResourcesLifecycle(t *testing.T) {
+	client, err := clients.NewDataArtsV1Client()
+	th.AssertNoErr(t, err)
+
+	workspace := ""
```
Comment (anton-sidelnikov) [inline acceptance/openstack/dataarts/v1/resources_test.go:30]: skip this tesk also

File: openstack/dataarts/v1/resource/Create.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	reqOpts := &golangsdk.RequestOpts{
+		OkCodes:     []int{200},
+		MoreHeaders: map[string]string{HeaderContentType: ApplicationJson},
+	}
+
+	if opts.Workspace != "" {
+		reqOpts.MoreHeaders[HeaderWorkspace] = opts.Workspace
+	}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1/resource/Create.go:50]: if it empty, should we return error here?

File: openstack/dataarts/v1/resource/Delete.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+func Delete(client *golangsdk.ServiceClient, resourceId, workspace string) error {
+	var err error
+	reqOpts := &golangsdk.RequestOpts{
+		MoreHeaders: map[string]string{HeaderContentType: ApplicationJson},
+		OkCodes:     []int{204},
+	}
+
+	if workspace != "" {
+		reqOpts.MoreHeaders[HeaderWorkspace] = workspace
+	}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1/resource/Delete.go:18]: and here, or not check at all

File: openstack/dataarts/v1/resource/Get.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+)
+
+// Get is used to query resource details. A resource contains various files such as JAR, ZIP, and properties files. A created resource can be used in job nodes such as DLI Spark and MRS Spark.
+// Send request GET /v1/{project_id}/resources/{resource_id}
+func Get(client *golangsdk.ServiceClient, resourceId, workspace string) (*Resource, error) {
+
+	var opts *golangsdk.RequestOpts
+	if workspace != "" {
+		opts.MoreHeaders = map[string]string{HeaderWorkspace: workspace}
+	}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1/resource/Get.go:15]: same as above

File: openstack/dataarts/v1/resource/List.go [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+func List(client *golangsdk.ServiceClient, opts *ListOpts, workspace string) (*ListResp, error) {
+	url, err := golangsdk.NewURLBuilder().WithEndpoints(resourcesEndpoint).WithQueryParams(&opts).Build()
+	if err != nil {
+		return nil, err
+	}
+
+	var reqOpts *golangsdk.RequestOpts
+	if workspace != "" {
+		reqOpts.MoreHeaders = map[string]string{HeaderWorkspace: workspace}
+	}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1/resource/List.go:25]: same

File: openstack/dataarts/v1/resource/Update.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	}
+
+	reqOpts := &golangsdk.RequestOpts{
+		MoreHeaders: map[string]string{HeaderContentType: ApplicationJson},
+		OkCodes:     []int{204},
+	}
+
+	if resource.Workspace != "" {
+		reqOpts.MoreHeaders = map[string]string{HeaderWorkspace: resource.Workspace}
+	}
```
Comment (anton-sidelnikov) [inline openstack/dataarts/v1/resource/Update.go:24]: same

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: openstack/dataarts/v1/resource/Create.go (hunk 2 above)
Comment (sgmv) [reply openstack/dataarts/v1/resource/Create.go:50]: Nope, this parameter is not necessary.

File: openstack/dataarts/v1/resource/Delete.go (hunk 3 above)
Comment (sgmv) [reply openstack/dataarts/v1/resource/Delete.go:18]: This parameter is not necessary.

File: openstack/dataarts/v1/resource/Get.go (hunk 4 above)
Comment (sgmv) [reply openstack/dataarts/v1/resource/Get.go:15]: answered

File: openstack/dataarts/v1/resource/List.go (hunk 5 above)
Comment (sgmv) [reply openstack/dataarts/v1/resource/List.go:25]: answered

File: openstack/dataarts/v1/resource/Update.go (hunk 6 above)
Comment (sgmv) [reply openstack/dataarts/v1/resource/Update.go:24]: answered

File: acceptance/openstack/dataarts/v1/resources_test.go (hunk 1 above)
Comment (sgmv) [reply acceptance/openstack/dataarts/v1/resources_test.go:30]: done

**Review by anton-sidelnikov — APPROVED** [review]

**Comment by anton-sidelnikov** [comment]
regate

**Review by artem-lifshits — APPROVED** [review]

**Pull request merged**